hw/xbox/xid-sdl.c

```diff
@@ -194,21 +194,22 @@ static const XIDDesc desc_xid_xbox_gamepad = {
 
 #define BUTTON_MASK(button) (1 << ((button) - GAMEPAD_DPAD_UP))
 
+SDL_HapticEffect effect;
+
 static void update_output(USBXIDState *s) {
 #ifdef FORCE_FEEDBACK
-    if (s->sdl_haptic == NULL) { return; }
-
-    SDL_HapticLeftRight effect = {
-        .type = SDL_HAPTIC_LEFTRIGHT,
-        .length = SDL_HAPTIC_INFINITY,
-        /* FIXME: Might be left/right inverted */
-        .large_magnitude = s->out_state.right_actuator_strength,
-        .small_magnitude = s->out_state.left_actuator_strength
-    };
+    if (s->sdl_haptic == NULL) {
+        return;
+    }
+
+    SDL_memset( &effect, 0, sizeof(SDL_HapticEffect) ); // 0 is safe default
+    effect.type = SDL_HAPTIC_LEFTRIGHT;
+    effect.leftright.length = SDL_HAPTIC_INFINITY;
+    effect.leftright.large_magnitude = s->out_state.right_actuator_strength; // x/32767 strength
```

Comment:
The comment is pretty bad. I'm also not sure about the range of `s->out_state.right_actuator_strength` etc.

We should definitely have a comment describing the Xbox and SDL range, but the current comment is confusing.